webapp/src/dialogs.tsx

```diff
@@ -193,7 +195,134 @@ function renderVersionLink(name: string, version: string, url: string) {
     </p>;
 }
 
+export function showPackageErrorDialogAsync(badPackages: EditorPackage[], removePackage: (id: string) => Promise<void>, openLegacyEditor?: () => void) {
+    core.confirmAsync({
+        header: lf("Extension Errors"),
+        hideCancel: true,
+        agreeLbl: lf("Ok"),
+        agreeClass: "positive",
+        jsx: <div>
+                <p>{pxt.Util.lf("The following extensions are preventing the project from compiling:")}</p>
+                <div className="ui relaxed divided list">
+                {
+                    badPackages.map(epkg => <PackageErrorListItem package={epkg} key={epkg.getPkgId()} removePackage={curryRemove(epkg.getPkgId())}></PackageErrorListItem>)
+                }
+            </div>
+            { openLegacyEditor ? renderEditorVersionMessage(openLegacyEditor) : undefined }
+        </div>
+    }).done();
+
+    function curryRemove(id: string) {
+        return () => {
+            coretsx.hideDialog();
+            return removePackage(id)
+        }
+    }
+}
+
+export function renderEditorVersionMessage(onClick: () => void) {
+    return <div className="ui message">
+        <p>{pxt.Util.lf("This project was made in an older version of the editor and may not be compatible with the latest version. To open the project in the old editor,")} <a onClick={onClick}>{pxt.Util.lf("Click here")}.</a></p>
```

Comment:
`Click here` -> `click here`?

Reply:
Concatenating sentence does not always translate properly. Can we have a button instead?

Reply:
The new text is two sentences:

> This project was made in an older version of the editor and may not be compatible with the latest version. Open the project in the old editor


The entire second sentence will be a link, so no string concatenation. @pelikhan is that better for localizing?